File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/BillingPeriod.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.time.Instant;

public final class BillingPeriod  {
    private final Instant start;
    private final Instant end;

    public BillingPeriod(Instant start, Instant end) {
        this.start = start;
        this.end = end;
    }

    public Instant getStart() {
        return start;
}

    public Instant start() {
        return start;
    }

    public Instant getEnd() {
        return end;
}

    public Instant end() {
        return end;
    }
}

File: apps/backend2/src/Common/Domain/ValueObjects/BillingPeriod.cs
namespace OffceOs.Domain.Common.ValueObjects;

/// <summary>
/// Represents a billing period with a start and end date.
/// </summary>
public readonly record struct BillingPeriod(DateTime Start, DateTime End)
{
    public bool IsExpired => DateTime.UtcNow >= End;
    public bool IsCurrent => DateTime.UtcNow >= Start && DateTime.UtcNow < End;
    public TimeSpan Duration => End - Start;

    public static BillingPeriod CurrentMonth()
    {
        var now = DateTime.UtcNow;
        var start = new DateTime(now.Year, now.Month, 1, 0, 0, 0, DateTimeKind.Utc);
        return new BillingPeriod(start, start.AddMonths(1));
    }
}
